inline-r/src/Language/R/Parser.hs

```diff
@@ -0,0 +1,327 @@
+-- |
+-- Copyright:   (c) 2016, AlphaSheets, Inc
+-- Stability:   Experimental
+-- Portability: Portable
+--
+-- "Parser" module provides high-level interface that allows to parse
+-- SEXP values into the complex Haskell data types.
```

Comment:
Might give some inspiration:
```
The "Parser" module provides an interface to deconstruct SEXP values and
to compose destructors, where cascades of cases would be necessary
otherwise.
```